axum/src/form.rs

```diff
@@ -71,22 +72,17 @@ where
 {
     type Rejection = FormRejection;
 
-    async fn from_request(req: Request<B>, state: &S) -> Result<Self, Self::Rejection> {
-        if req.method() == Method::GET {
-            let query = req.uri().query().unwrap_or_default();
-            let value = serde_urlencoded::from_str(query)
-                .map_err(FailedToDeserializeQueryString::__private_new)?;
-            Ok(Form(value))
-        } else {
-            if !has_content_type(req.headers(), &mime::APPLICATION_WWW_FORM_URLENCODED) {
-                return Err(InvalidFormContentType.into());
+    async fn from_request(req: Request<B>, _state: &S) -> Result<Self, Self::Rejection> {
+        match req.extract().await {
+            Ok(RawForm(bytes)) => {
+                let value = serde_urlencoded::from_bytes(&bytes)
+                    .map_err(FailedToDeserializeQueryString::__private_new)?;
```

Comment:
Hm why is this named `FailedToDeserializeQueryString` again? (since this can be from the body)

Reply:
Yeah its a little odd. Maybe `FailedToDeserializeForm`?

Reply:
Yeah sounds better.